book/tuto-08-gouraud.md

```diff
@@ -42,10 +42,12 @@ If the light is *parallel* to the normal, then the surface should be bright. And
 ## In practice
 
 The main part of the calculation will be done in the fragment shader. However we need to modify
-the vertex shader first, in order to pass the normal's data to the fragment shader.
+the vertex shader first, in order to pass the normal's data to the fragment shader. In addition
+to this, also note that some hardware doesn't support some functions that we're going to use in
```

Comment:
It's not "some hardware that doesn't support some functions". It's just that GLSL v140 itself doesn't support these functions.